isubit/suitcase
# LUGG-828 Universal Grid Styling

## What is it?

This pull request implements some simple jQuery and CSS to create a grid for views.
## Why?

Clients were asking for grid styling for a few content types, such as resources and videos. When resources were implemented in the past, the targeting was too specific and any cloned view would not be grid-ified. These changes seek to fill that need by creating styling that is easy to access by simply adding a couple of classes in views (which we were already doing with resources and people). Another benefit of this pull request is that any view that uses the grid styling will utilize the fields format in views, which is much easier to manipulate programmatically than the display suite format.
## Changes
- Removed commented-out CSS from suitcase.css
- Added simple jQuery that matches grid cell heights
- Removed grid styling from luggage_people and luggage_resources and moved it to suitcase.css
## Testing

These steps assume that you will be using a local copy of the entomology website.
- Pull down these changes
- Clear caches (I also run the drush command registry-rebuild, but that might be overkill)
- Go to edit the video view
- Change the video view to an unformatted list with fields
- Edit the unformatted list settings and add luggage-gridrow as a class to the page
- Add luggage-grid as a class under Advanced View settings > Other > CSS Class.
- Remove the weird Title & Body field that’s automatically added to the view and make sure that title and body are being displayed. Rearrange the field order to video, title, then body.
- Save the view
- Look at the video view to see that it is a grid of videos
- Resize the window and make sure that the cells resize accordingly. They should fit the whole window when it is smaller than ~739 pixels wide, text should not be cut off, each row should resize according to the highest cell in that row, no funky text overflow.
## Questions I have
- Do we keep the green text, such as in people position and resource URLs? If you look at the suitcase/luggage site, it isn’t in use on the people view. Seems like a possible issue for colorblind users and I don’t like how it applies to people position (not a link) and resource URL (is a clickable link). Not consistent
- Is the way I added padding before body text okay? Adding padding before the p tag seems hackish, but adding padding after the title view field was waaaayyy too much white space there.
